File: BookStoreApi/Database/Models/Invoice.cs
using BookStoreApi.Enums;

namespace BookStoreApi.Database.Models
{
    public class Invoice
    {
        public int Id { get; set; }
        public long TotalPrice { get; set; }
        public long Maliat { get; set; }
        public long FinalTotalPrice { get; set; }
        public PaymentStatus PaymentStatus { get; set; }
        public InvoiceStatus InvoiceStatus { get; set; }
        public int UserId { get; set; }
        public int AddressId { get; set; }
        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public DateTime? ValidatedAt { get; set; }
        public List<Book>? Books { get; set; }
        public List<Payment>? Payments { get; set; }
        public User? User { get; set; }
        public AddressInfo? Address { get; set; }
    }
}

File: BookStoreFront/src/app/models/invoice.ts
import { Address } from "./address";
import { PaginationInfo } from "./apiResponse";
import { BookAllData } from "./book";
import { InvoiceStatus, PaymentStatus } from "./enum";
import { Payment } from "./payment";
import { User } from "./user";

export interface Invoice {
  id: number;
  totalPrice: number;
  maliat: number;
  finalTotalPrice: number;
  paymentStatus: PaymentStatus;
  invoiceStatus: InvoiceStatus;
  userId: number;
  addressId: number;
  createdAt?: Date;
  updatedAt?: Date;
  validatedAt?: Date;

  books?: BookAllData[] | null;
  payments?: Payment[] | null;
  user?: User | null;
  address?: Address | null;
}

export interface UserInvoicesList {
  invoices: Invoice[] | null;
  pagination: InvoicePaginationInfo | null;
}

export interface CreateInvoiceRequest {
  addressId: number;
  books: number[]; // Array of book IDs
  counts: number[]; // Array of quantities
}

export interface InvoicePaginationInfo extends PaginationInfo {}